src/raftstore/store/worker/apply.rs

```diff
@@ -1623,91 +1718,151 @@ impl ApplyDelegate {
+        )?;
+
+        self.cleanup_range_cf(ctx, cf, &start_key, &end_key);
+
+        ranges.push(Range::new(cf.to_owned(), start_key, end_key));
+
+        Ok(Response::new())
+    }
+
+    fn handle_ingest_sst(
+        &mut self,
+        ctx: &ApplyContext,
+        req: &Request,
+        ssts: &mut Vec<SSTMeta>,
+    ) -> Result<Response> {
+        assert!(!self.is_pause());
+
+        let sst = req.get_ingest_sst().get_sst();
+
+        if let Err(e) = check_sst_for_ingestion(sst, &self.region) {
+            error!("ingest {:?} to region {:?}: {:?}", sst, self.region, e);
+            // This file is not valid, we can delete it here.
+            let _ = ctx.importer.delete(sst);
+            return Err(e);
+        }
+
+        ctx.importer.ingest(sst, &self.engine).unwrap_or_else(|e| {
+            // If this failed, it means that the file is corrupted or something
+            // is wrong with the engine, but we can do nothing about that.
+            panic!("{} ingest {:?}: {:?}", self.tag, sst, e);
+        });
+
+        ssts.push(sst.clone());
+        Ok(Response::new())
+    }
+
+    fn convert_range(&self, s_key: &[u8], e_key: &[u8]) -> Result<(Vec<u8>, Vec<u8>)> {
+        // start key must less than end key, the maximum end key is empty.
         if !e_key.is_empty() && s_key >= e_key {
             return Err(box_err!(
-                "invalid delete range command, start_key: {:?}, end_key: {:?}",
+                "invalid start_key: {:?}, end_key: {:?}",
                 s_key,
                 e_key
             ));
         }
+
+        // check start key
         check_data_key(s_key, &self.region)?;
+        let start_key = keys::data_key(s_key);
+
+        // check end key
         let end_key = keys::data_end_key(e_key);
         let region_end_key = keys::data_end_key(self.region.get_end_key());
         if end_key > region_end_key {
             return Err(Error::KeyNotInRegion(e_key.to_vec(), self.region.clone()));
         }
 
-        let resp = Response::new();
-        let mut cf = req.get_delete_range().get_cf();
+        Ok((start_key, end_key))
+    }
+
+    fn convert_cf(cf: &str) -> Result<&str> {
         if cf.is_empty() {
-            cf = CF_DEFAULT;
-        }
-        if ALL_CFS.iter().find(|x| **x == cf).is_none() {
-            return Err(box_err!("invalid delete range command, cf: {:?}", cf));
+            Ok(CF_DEFAULT)
+        } else if ALL_CFS.iter().find(|x| **x == cf).is_none() {
+            Err(box_err!("invalid unsafe cleanup command, cf: {:?}", cf))
+        } else {
+            Ok(cf)
         }
-        let handle = rocksdb::get_cf_handle(&self.engine, cf).unwrap();
+    }
 
-        let start_key = keys::data_key(s_key);
-        // Use delete_files_in_range to drop as many sst files as possible, this
-        // is a way to reclaim disk space quickly after drop a table/index.
+    fn cleanup_range_cf(&mut self, ctx: &ApplyContext, cf: &str, start_key: &[u8], end_key: &[u8]) {
```

Comment:
I think this is duplicate with `unsafe_cleanup_range_cf` in `sub_runner`, why not extract a function to handle both `delete files` and `delete range`?